lib/gems/pending/util/xml/miq_rexml.rb

```diff
@@ -148,7 +148,7 @@ def initialize(first, second = nil, parent = nil)
         rescue => err
           if err.class == ::Encoding::CompatibilityError
             second_utf8 = second.to_s.force_encoding('UTF-8')
```

Comment:
Shouldn't this just be changed to something like this:
```ruby
second_utf8 = second.to_s.encode('UTF-8', :invalid => :replace, :undef => :replace, :replace => "")
```
The string may not be UTF-8, so you don't want to force the encoding. You want to transcode the string to UTF-8, removing the characters that can't be transcoded.

Reply:
I tested the above replacement. It didn't work for our case. The final result still has unacceptable chars. The test string I used is the 1st partition type: \xAF=\xC6\x0F\x83\x84rG\x8Ey=i\xD8G}\xE4.

irb(main)> "\xAF=\xC6\x0F\x83\x84rG\x8Ey=i\xD8G}\xE4".encode('UTF-8', :invalid => :replace, :undef => :replace, :replace => "")
=> "=\u000FrGy=iG}"

Reply:
What makes you say it contains unacceptable characters? If you transcoded the string as I suggested, the result has to be a valid UTF-8 string. Did `vaild_encoding?` on the resultant string return false?

Reply:
The valid_encoding? return true. But somehow we still got error message: incompatible encoding regexp match (UTF-8 regexp with ASCII-8BIT string) in evm.log